coap-gateway/service/service.go

```diff
@@ -211,6 +213,15 @@ func New(ctx context.Context, config Config, logger *zap.Logger) (*Service, erro
 		}
 	}
 
+	provider, err := provider.NewPlgdProvider(config.OAuthDeviceClient,
+		logger,
+		config.Clients.AuthServer.OwnerClaim,
+		"query", "offline", "code")
+	if err != nil {
+		resourceSubscriber.Close()
+		return nil, fmt.Errorf("cannot create device provider: %w", err)
+	}
+
```

Comment:
add line with`resourceSubscriber.AddCloseFunc(provider.Close)` - it may solve issue with FD